### PR TITLE
[master] Add wildcard support for service generation

### PR DESCRIPTION
**Pull request by aneeshafedo**
Resolves https://github.com/ballerina-platform/openapi-tools/issues/628

## Goals
**Sample OpenAPI**

```yaml
  /pets/my:
    post:
      summary: Create a pet
      operationId: createPet
      tags:
        - pets
      requestBody:
        description: "Request to add a pet"
        content:
          '*/*':
            schema:
              $ref: "#/components/schemas/Pets"
      responses:
        '200':
          description: Successful operation
          content:
            '*/*':
              schema:
                type: array
                items:
                  description: "Pet records"
```

**Generated Ballerina resource**

service.bal

```bal
import ballerina/http;

listener http:Listener ep0 = new (80, config = {host: "petstore.openapi.io"});

service /v1 on ep0 {
    resource function post pets/my(@http:Payload http:Request request) returns OkAnydata {
    }
}
```
types.bal

```bal
public type OkAnydata record {|
    *http:Ok;
    anydata body;
|};
...
```

### CODE REVIEW

File: openapi-core/src/main/java/io/ballerina/openapi/core/generators/service/RequestBodyGenerator.java [hunk 1]
```diff
@@ -170,8 +172,14 @@ null, createIdentifierToken(GeneratorConstants.BYTE)),
...
                 default:
-                    identifierToken = createIdentifierToken(GeneratorConstants.JSON);
-                    typeName = Optional.ofNullable(createSimpleNameReferenceNode(identifierToken));
+                    ImmutablePair<Optional<TypeDescriptorNode>, Optional<TypeDefinitionNode>> mediaTypeTokens =
+                            handleMediaType(mediaType, null);
+                    if (mediaTypeTokens.left.isEmpty()) {
+                        identifierToken = createIdentifierToken(HTTP_REQUEST);
+                        typeName = Optional.ofNullable(createSimpleNameReferenceNode(identifierToken));
+                    } else {
+                        typeName = mediaTypeTokens.left;
```
Comment (NipunaRanasinghe) [inline openapi-core/src/main/java/io/ballerina/openapi/core/generators/service/RequestBodyGenerator.java:181]:
```suggestion
                        typeName = mediaTypeTokens.getLeft();
```

Comment (NipunaRanasinghe) [inline openapi-core/src/main/java/io/ballerina/openapi/core/generators/service/RequestBodyGenerator.java:181]: Shall we swap the if else order in here, as we usually handle the happy path within the if case?

Comment (aneeshafedo) [reply openapi-core/src/main/java/io/ballerina/openapi/core/generators/service/RequestBodyGenerator.java]: Updated

Comment (aneeshafedo) [reply openapi-core/src/main/java/io/ballerina/openapi/core/generators/service/RequestBodyGenerator.java]: Updated

**Review by NipunaRanasinghe — APPROVED** [review]

**Review by lnash94 — APPROVED** [review]

**Pull request merged**